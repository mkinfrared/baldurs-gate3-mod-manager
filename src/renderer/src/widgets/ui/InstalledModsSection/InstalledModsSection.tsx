import { Suspense } from "react";

import { InstalledModsList } from "@renderer/features/modSettingsFile";

import css from "./InstalledModsSection.module.scss";
import { InstalledModsSectionProps } from "./InstalledModsSection.type";

const InstalledModsSection = ({ className }: InstalledModsSectionProps) => {
  const classNames = [css.InstalledModsSection, className]
    .filter(Boolean)
    .join(" ");

  return (
    <div className={classNames} data-testid="InstalledModsSection">
      <Suspense fallback={<div>Loading..</div>}>
        <InstalledModsList />
      </Suspense>
    </div>
  );
};

export { InstalledModsSection };
