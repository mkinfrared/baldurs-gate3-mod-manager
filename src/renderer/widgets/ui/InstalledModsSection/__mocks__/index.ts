export { default } from "./InstalledModsSection.mock";
