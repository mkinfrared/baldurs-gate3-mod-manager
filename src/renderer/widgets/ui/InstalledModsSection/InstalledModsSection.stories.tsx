import { Meta, StoryObj } from "@storybook/react";

import { InstalledModsSection } from "./InstalledModsSection";
import css from "./Story.module.scss";

type Story = StoryObj<typeof meta>;

const meta = {
  title: "UI/InstalledModsSection",
  component: InstalledModsSection,
  tags: ["autodocs"],
} satisfies Meta<typeof InstalledModsSection>;

const Default: Story = {
  args: {},
  render: (args) => (
    <div className={css.Story}>
      <InstalledModsSection {...args} />
    </div>
  ),
};

export { Default };

export default meta;
