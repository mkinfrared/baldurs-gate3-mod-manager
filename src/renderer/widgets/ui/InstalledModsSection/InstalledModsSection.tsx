import { Suspense } from "react";

import { InstalledModsList } from "renderer/features/modSettingsFile";
import { Heading } from "renderer/shared/ui";

import css from "./InstalledModsSection.module.scss";
import { InstalledModsSectionProps } from "./InstalledModsSection.type";

const InstalledModsSection = ({ className }: InstalledModsSectionProps) => {
  const classNames = [css.InstalledModsSection, className]
    .filter(Boolean)
    .join(" ");

  return (
    <div className={classNames} data-testid="InstalledModsSection">
      <Heading>Installed Mods</Heading>
      <Suspense fallback={<div>Loading..</div>}>
        <InstalledModsList />
      </Suspense>
    </div>
  );
};

export { InstalledModsSection };
