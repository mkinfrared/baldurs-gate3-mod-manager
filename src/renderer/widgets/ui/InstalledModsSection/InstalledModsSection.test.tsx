import { render } from "@testing-library/react";

import { InstalledModsSection } from "./InstalledModsSection";

jest.mock("renderer/features/modSettingsFile/ui/InstalledModsList");

describe("<InstalledModsSection />", () => {
  const Component = <InstalledModsSection />;

  it("should be defined", () => {
    expect(InstalledModsSection).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });

  it("should contain a data test id", () => {
    const { getByTestId } = render(Component);
    const element = getByTestId("InstalledModsSection");

    expect(element).toBeDefined();
  });
});
