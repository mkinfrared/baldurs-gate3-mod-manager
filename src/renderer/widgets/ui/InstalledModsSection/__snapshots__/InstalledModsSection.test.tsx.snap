// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InstalledModsSection /> should match the snapshot 1`] = `
<div>
  <div
    class="InstalledModsSection"
    data-testid="InstalledModsSection"
  >
    <div
      class="InstalledModsList"
      data-testid="InstalledModsList"
    >
      <div>
        <h3
          class="Heading"
          data-testid="Heading"
        >
          Active Mods
        </h3>
        <div
          class="Card EmptyList"
          data-testid="Card"
        >
          <h4
            class="Heading"
            data-testid="Heading"
          >
            Activate the mods from the Installed Mods list below using Activate button and they will appear here
          </h4>
        </div>
      </div>
      <div>
        <h3
          class="Heading"
          data-testid="Heading"
        >
          Installed Mods
        </h3>
        <div
          class="Card EmptyList"
          data-testid="Card"
        >
          <h4
            class="Heading"
            data-testid="Heading"
          >
            You haven't installed any mods yet. Use the panel on the right to install mods
          </h4>
        </div>
      </div>
    </div>
  </div>
</div>
`;
